client/me/purchases/manage-purchase/payment-method-selector/assignment-processor-functions.ts

```diff
@@ -158,7 +158,7 @@ interface NewCardSubmitData {
 
 export async function assignExistingCardProcessor(
 	purchase: Purchase | undefined,
-	reduxDispatch: ReturnType< typeof useDispatch >,
+	reduxDispatch: CalypsoDispatch,
 	submitData: unknown
 ): Promise< PaymentProcessorResponse > {
 	recordFormSubmitEvent( { reduxDispatch, purchase } );
```

Comment:
This could probably be changed to a pure action creator also, although it's not super important.

Reply:
Converted in b764fb524e. Actually all functions that accept `reduxDispatch` as a parameter could and should be thunks, but there are too many of them.

By the way, seems to me this `recordFormSubmitEvent` action behaves incorrectly when the `useForAllSubscriptions` param is not passed. Then the tracks event is recorded with a `use_for_all_subs: 'undefined'` field. @sirbrillig I'm leaving this up to you to fix, because I don't fully understand this code and wanted to avoid behavior changes in this PR.